tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/settings/JmapSettingsRepository.scala

```diff
@@ -0,0 +1,73 @@
+package com.linagora.tmail.james.jmap.settings
+
+import com.google.common.base.Preconditions
+import com.google.common.collect.{HashBasedTable, Table, Tables}
+import com.linagora.tmail.james.jmap.settings.JmapSettings.{JmapSettingsKey, JmapSettingsValue}
+import com.linagora.tmail.james.jmap.settings.JmapSettingsStateFactory.INITIAL
+import javax.inject.Inject
+import org.apache.james.core.Username
+import org.apache.james.jmap.core.UuidState
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.SMono
+
+import scala.collection.mutable
+
+trait JmapSettingsRepository {
+  def get(username: Username): Publisher[JmapSettings]
+
+  def save(username: Username, settings: JmapSettingsUpsertRequest): Publisher[SettingsStateUpdate]
+
+  def reset(username: Username, settings: JmapSettingsUpsertRequest): Publisher[SettingsStateUpdate]
```

Comment:
What is the difference between `save` and `reset` ?

Reply:
reset will remove all redundant settings, save will not

Reply:
And does it make sense, is it usefull? IMO we just need `reset` being exposed by the API